test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpImplicitExpressionTest.cs

```diff
@@ -17,6 +18,96 @@ public override ParserBase CreateCodeParser()
+                    };
+
+                // implicitExpression, expectedImplicitExpression, acceptedCharacters, expectedErrors
+                return new TheoryData<string, string, AcceptedCharacters, RazorError[]>
+                {
+                    { "val??[", "val", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val??[0", "val", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[", "val?[", AcceptedCharacters.Any, missingEndBracketError(5) },
+                    { "val?(", "val", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[more", "val?[more", AcceptedCharacters.Any, missingEndBracketError(5) },
+                    { "val?[0]", "val?[0]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[<p>", "val?[", AcceptedCharacters.Any, missingEndBracketError(5) },
+                    { "val?[more.<p>", "val?[more.", AcceptedCharacters.Any, missingEndBracketError(5) },
+                    { "val??[more<p>", "val", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[-1]?", "val?[-1]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[abc]?[def", "val?[abc]?[def", AcceptedCharacters.Any, missingEndBracketError(11) },
+                    { "val?[abc]?[2]", "val?[abc]?[2]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[abc]?.more?[def]", "val?[abc]?.more?[def]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[abc]?.more?.abc", "val?[abc]?.more?.abc", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[null ?? true]", "val?[null ?? true]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                    { "val?[abc?.gef?[-1]]", "val?[abc?.gef?[-1]]", AcceptedCharacters.NonWhiteSpace, noErrors },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NullConditionalOperatorData_Bracket))]
+        public void ParseBlockMethodParsesNullConditionalOperatorImplicitExpression_Bracket(
+            string implicitExpresison,
+            string expectedImplicitExpression,
+            AcceptedCharacters acceptedCharacters,
+            RazorError[] expectedErrors)
+        {
+            // Act & Assert
+            ImplicitExpressionTest(
+                implicitExpresison,
+                expectedImplicitExpression,
+                acceptedCharacters,
+                expectedErrors);
+        }
+
+        public static TheoryData NullConditionalOperatorData_Dot
+        {
+            get
+            {
+                // implicitExpression, expectedImplicitExpression
+                return new TheoryData<string, string>
+                {
+                    { "val?", "val" },
+                    { "val??", "val" },
+                    { "val??more", "val" },
```

Comment:
Not something that should change in this PR but this case is interesting. Does the test mean null coalescing expressions aren't supported in implicit expressions, at least outside brackets or parentheses? Might be worth a bug since this case is unlikely to be an typo e.g. would expect whitespace after `??` or `???` if this were English.